.github/CODEOWNERS

```diff
@@ -114,6 +114,8 @@
 /libs/application/template-api-modules/src/lib/modules/templates/public-debt-payment-plan     @island-is/sendiradid
 /libs/api/domains/health-insurance/                                                           @island-is/sendiradid
 /libs/application/templates/health-insurance/                                                 @island-is/sendiradid
+/libs/clients/rsk-company-info/                                                               @island-is/sendiradid
+/libs/api/domains/rsk-company-info/                                                           @island-is/sendiradid
```

Comment:
1. Unfortunately there are two APIs from skatturinn for company information, so there are two corresponding clients, rsk and rsk-company-info.

2. I don't think there is no particular reason why they are being kept seperate, except maybe since there were two clients there were made two domains. They have just now been merged into one domain rsk

Reply:
Regarding 1. Not sure if I understand you correctly.
You're saying there are two APIs from skatturinn for company information. This PR is implementing one of them.
Is the other version inside `clients/rsk/v(1|2)`?

Reply:
@darrikonn yes I believe so

Reply:
So IMO it makes a total sense that this is just another version of the `rsk` client, thus it should live inside `clients/rsk/v3`

Reply:
I took a closer look at what the other API and they don't exactly provide the same information. One is for fetching all companies that a kennitala has access to, and the other provide information about a company kennitala (companyName, registraitonDate and more). They also have separate base URLs.  Personally I feel like this isn't really a new version unless there exists a new API from Skatturinn that includes the previous functions (since the function from v2 is not being deprecated, or at least now that I know of).

@darrikonn Thoughts?

Reply:
Tó complicate this further, some of the existing clients are for APIs which should not be used anymore (Vidspyrna related). RSK has created yet another API for getting all companies that a kennitala has access to, which I need to integrate.

So how should all of these be named?

IMO, we should have a separate client lib for each published API we can integrate with. It should be named according to the published API name, but they can be grouped by provider (eg libs/clients/rsk/company-registry). If a publisher publishes two versions of the same API, I would prefer to keep the clients in one library if we can make sure the interface is backwards compatible, but this can sometimes be tricky.

However, when it comes to GraphQL domains, we should try to “simplify” the world. Clients should not map to domains 1-to-1. A domain might pull data from multiple APIs when that makes sense. Basically, domains should be domain-oriented. This is of course very subjective and it’s tricky to design the GraphQL schema in a distributed and consistent way. I don’t have any solutions yet, but would appreciate a schema design sessions in each new integration. Unfortunately many integrations are quite focused (eg for a one-off application) which complicates things further.

IMHO, this specific integration might be represented in some focused CompanyRegistry domain, but it should also be pulled in to the Identity domain.

Reply:
Yeah I see where you're going with this. This is really a question of how we look at this.

1. `v1` | `v2` | `v3` are just three different versions (not necessarily related) of a client, and should not be thought of as the increased number being an improved version. So `v2` is just another version and might contain other endpoints that provide different informations than `v3`.
2. `v1` | `v2` | `v3` are all versions of the exact same client and can be thought of as the higher number is an improved version.

I originally interpreted it as `1`, but seems like you're going with `2`; if I understand you correctly.

It's probably best to generate a general discussion about this on Slack in #dev_faghopur.

Reply:
Yeah I agree @eirikurn .. like to group it by the provider and name them according to the published API name (so stop using `v1`-`vx`, unless we can't integrate and have to go for e.g. `/libs/clients/rsk/company-registry-v1`).

"Clients should not map to domains 1-to-1." -> Definitely!

Reply:
So we’re saying that the Client should be: `/libs/clients/rsk/company-registry

And the domain should be? CompanyRegistry, with companyRegistryCompany query? :)